### DISCUSSION TITLE
V5版本使用官方示例出现报错

### DISCUSSION THREAD

**Discussion by recursively** [antvis/G6#6301] [Q&A] [answered]
使用官方示例并使用“force-atlas2”等布局时，存在报错，直接无法渲染出来：

<img width="530" alt="image" src="https://github.com/user-attachments/assets/1cb688ed-f001-4f35-8deb-ea15f0297bf9">

使用代码为官方示例代码：

import { Graph } from '@antv/g6';

fetch('https://gw.alipayobjects.com/os/antvdemo/assets/data/relations.json')
  .then((res) => res.json())
  .then((data) => {
    const graph = new Graph({
      container: 'container',
      data,
      autoFit: 'view',
      layout: {
        type: 'force-atlas2',
        preventOverlap: true,
        kr: 20,
        center: [250, 250],
      },
      behaviors: ['zoom-canvas', 'drag-canvas'],
      autoResize: true,
      zoomRange: [0.1, 5],
    });

    graph.render();
  });

目前测试下来只有d3-force布局不报错，我是直接在v4基础上直接安装v5进行升级的，是否跟这个有关？

**Comment by resetsix** [comment]
> 使用官方示例并使用“force-atlas2”等布局时，存在报错，直接无法渲染出来：
> <img alt="image" width="530" src="https://private-user-images.githubusercontent.com/39149024/365540567-1cb688ed-f001-4f35-8deb-ea15f0297bf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzE3NjUsIm5iZiI6MTcyNTg3MTQ2NSwicGF0aCI6Ii8zOTE0OTAyNC8zNjU1NDA1NjctMWNiNjg4ZWQtZjAwMS00ZjM1LThkZWItZWExNWYwMjk3YmY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDA4NDQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkYjkyYzA3NjUyMzMzMmYwMTU5ZGNhZjFhNjQ2MmU4MDI3MzZkMzY2Yzk0MGVjOGQwNzUyNWM4MGZjYjA2YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I1WdWWhzQtOZU1AKNliYVbRpjAXZnRBIH3YLZrCvH9g">
> 
> 使用代码为官方示例代码：
> 
> import { Graph } from '@antv/g6';
> 
> fetch('https://gw.alipayobjects.com/os/antvdemo/assets/data/relations.json') .then((res) => res.json()) .then((data) => { const graph = new Graph({ container: 'container', data, autoFit: 'view', layout: { type: 'force-atlas2', preventOverlap: true, kr: 20, center: [250, 250], }, behaviors: ['zoom-canvas', 'drag-canvas'], autoResize: true, zoomRange: [0.1, 5], });
> 
> ```
> graph.render();
> ```
> 
> });
> 
> 目前测试下来只有d3-force布局不报错，我是直接在v4基础上直接安装v5进行升级的，是否跟这个有关？

官网文档在线运行环境直接修改参数可能会有点问题，尤其是不同版本 api 已经变化了很多。

在本地跑 demo 试试。

**Reply by recursively** [reply]
> > 使用官方示例并使用“force-atlas2”等布局时，存在报错，直接无法渲染出来：
> > <img alt="image" width="530" src="https://private-user-images.githubusercontent.com/39149024/365540567-1cb688ed-f001-4f35-8deb-ea15f0297bf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzE3NjUsIm5iZiI6MTcyNTg3MTQ2NSwicGF0aCI6Ii8zOTE0OTAyNC8zNjU1NDA1NjctMWNiNjg4ZWQtZjAwMS00ZjM1LThkZWItZWExNWYwMjk3YmY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDA4NDQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkYjkyYzA3NjUyMzMzMmYwMTU5ZGNhZjFhNjQ2MmU4MDI3MzZkMzY2Yzk0MGVjOGQwNzUyNWM4MGZjYjA2YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I1WdWWhzQtOZU1AKNliYVbRpjAXZnRBIH3YLZrCvH9g">
> > 使用代码为官方示例代码：
> > import { Graph } from '@antv/g6';
> > fetch('https://gw.alipayobjects.com/os/antvdemo/assets/data/relations.json') .then((res) => res.json()) .then((data) => { const graph = new Graph({ container: 'container', data, autoFit: 'view', layout: { type: 'force-atlas2', preventOverlap: true, kr: 20, center: [250, 250], }, behaviors: ['zoom-canvas', 'drag-canvas'], autoResize: true, zoomRange: [0.1, 5], });
> > ```
> > graph.render();
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > });
> > 目前测试下来只有d3-force布局不报错，我是直接在v4基础上直接安装v5进行升级的，是否跟这个有关？
> 
> 官网文档在线运行环境直接修改参数可能会有点问题，尤其是不同版本 api 已经变化了很多。
> 
> 在本地跑 demo 试试。


就是在本地运行的，并不是在官网的playground里运行的，只不过使用的是上述官网提供的代码，如果官网代码存在问题，能否提供一下可以跑的代码示例呢？谢谢。现在测试下来只有d3-force这个布局不会报错，不知道是什么原因

**Comment by resetsix** [comment, marked as answer]
官网`Force Atlas 2`示例（正常）：https://g6-next.antv.antgroup.com/zh/examples/layout/force-directed/#atlas2

我在`(antv/g6 v5.0.20)`本地使用`force-atlas2`布局也没任何 warning信息。

排查思路：

1、重装依赖。
```bash
## 使用 npm 重新安装 @antv/g6
npm uninstall @antv/g6
npm install @antv/g6

## 使用 yarn 重新安装 @antv/g6
yarn remove @antv/g6
yarn add @antv/g6
```

**Reply by recursively** [reply]
确实是依赖的问题，重装一下解决了，感谢~
